MultiClient/Archipelago.MultiClient.Net/ArchipelagoSession.cs

```diff
@@ -17,32 +17,40 @@ public class ArchipelagoSession
         public delegate void ErrorReceivedHandler(Exception e, string message);
         public event ErrorReceivedHandler ErrorReceived;
 
+        public delegate void SocketClosedHandler(CloseEventArgs e);
+        public event SocketClosedHandler SocketClosed;
+
         public string Url { get; private set; }
-        public bool Connected = false;
+        public bool Connected { get => Socket.IsAlive; }
```

Comment:
I think the issue here is that IsAlive is not async! So this will stall the game if you are not local. That's why I had it as a separate bool.

Reply:
This is correct, from what I can tell. Any call to get `Socket.IsAlive` sends a ping to the websocket host and waits max 1 second, by default, for a pong. Any calls to check should keep that in mind.